components/board/client.go

```diff
@@ -232,6 +249,29 @@ func (arc *analogReaderClient) Read(ctx context.Context, extra map[string]interf
 	return int(resp.Value), nil
 }
 
+// analogWriterClient satisfies a gRPC based board.AnalogWriter. Refer to the interface
+// for descriptions of its methods.
+type analogWriterClient struct {
+	*client
+	boardName string
+	pinName   string
+}
+
+func (awc *analogWriterClient) Write(ctx context.Context, value int32, extra map[string]interface{}) error {
+	ext, err := protoutils.StructToStructPb(extra)
+	if err != nil {
+		return err
+	}
+	_, err = awc.client.client.WriteAnalog(ctx, &pb.WriteAnalogRequest{
+		Name:  awc.boardName,
```

Comment:
why do we need a board name?

Reply:
we need it for the call to the server to find the board using the board name input, and then call `board.AnalogWriterByName(pin)` to get the analog writer to call Write